src/relayFeeCalculator/relayFeeCalculator.ts

```diff
@@ -198,13 +201,38 @@ export class RelayFeeCalculator {
     return this.queries.getTokenPrice(tokenSymbol);
   }
 
-  async gasFeePercent(amountToRelay: BigNumberish, tokenSymbol: string, _tokenPrice?: number): Promise<BigNumber> {
+  async gasFeePercent(
+    amountToRelay: BigNumberish,
+    tokenSymbol: string,
+    originChainId: number,
+    destinationChainId: number,
+    messagePayload?: {
+      message: string;
+      recipientAddress: string;
+    },
```

Comment:
This has been added like this to enforce a mutual requirement. Either we don't include any of these values (and default to the normal gas estimation) or we include `message` and are forced to include the recipient.

Reply:
I'm not sure just these should be a mutual requirement. For example, would there really ever be a reason why we'd have `recipientAddress` be undefined? Some of the other parameters seem just as required like the chain.

Reply:
Hmm, that's fair, but we didn't require them before. Since we're essentially falling back to previous logic when we don't include the message, should we enforce the recipient? We would need to change the calling references in the FE as well.

Reply:
I do think we might need to introduce a breaking change to this interface and require `recipientAddress` and `message` going forward. What do you think long term is more scalable to better support clients?

Reply:
Agreed. As long as we are okay making a breaking change, I think requiring a recipient address and making the message default to empty would be the best interface going forward.

We could do some defaulting behavior or make them optional, but a required recipient will probably produce more accurate results and require less branching/casework.

Reply:
Done: https://github.com/across-protocol/sdk-v2/pull/427/commits/d94450934f46b17d6fdd0a9ce687378490b4bb64